src/DIPS.Xamarin.UI/Controls/Popup/PopupBehaviour.cs

```diff
@@ -0,0 +1,127 @@
+            }
+
+            base.OnAttachedTo(bindable);
+        }
+
+        protected override void OnDetachingFrom(View bindable)
+        {
+            base.OnDetachingFrom(bindable);
+        }
+
+        public static readonly BindableProperty CloseOnClickProperty =
+            BindableProperty.CreateAttached("CloseOnClick", typeof(bool), typeof(PopupBehaviour), false, propertyChanged: OnCloseOnClickChanged);
+
+        public static void OnCloseOnClickChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            var view = (View)bindable;
+            view.BindingContextChanged += (s, e) =>
+            {
+                var popupLayout = view.GetParentOfType<PopupLayout>();
+                if (popupLayout != null)
+                {
+                    popupLayout.AddOnCloseRecognizer(view);
+                }
+            };
+        }
+
+        public static void SetCloseOnClick(BindableObject view, bool value)
+        {
+            view.SetValue(CloseOnClickProperty, value);
+        }
+
+        public static bool GetCloseOnClick(BindableObject view)
+        {
+            return (bool)view.GetValue(CloseOnClickProperty);
+        }
+
+        public static readonly BindableProperty PopupBindingContextFactoryProperty =
+            BindableProperty.Create(nameof(PopupBindingContextFactory), typeof(Func<object>), typeof(PopupBehaviour), null);
+
+        public Func<object> PopupBindingContextFactory
+        {
+            get { return (Func<object>)GetValue(PopupBindingContextFactoryProperty); }
+            set { SetValue(PopupBindingContextFactoryProperty, value); }
+        }
+
+        public static readonly BindableProperty SaveCommandProperty =
+            BindableProperty.Create(nameof(SaveCommand), typeof(ICommand), typeof(PopupBehaviour), null);
+
+        public ICommand SaveCommand
+        {
+            get { return (ICommand)GetValue(SaveCommandProperty); }
+            set { SetValue(SaveCommandProperty, value); }
+        }
+
+        public static readonly BindableProperty SaveTextProperty =
+            BindableProperty.Create(nameof(SaveText), typeof(string), typeof(PopupBehaviour), null);
+
+        public string SaveText
+        {
+            get { return (string)GetValue(SaveTextProperty); }
+            set { SetValue(SaveTextProperty, value); }
+        }
+
+        public static readonly BindableProperty DirectionProperty =
+            BindableProperty.Create(nameof(Direction), typeof(PopupDirection), typeof(PopupBehaviour), PopupDirection.Auto);
+
+        public PopupDirection Direction
+        {
+            get { return (PopupDirection)GetValue(DirectionProperty); }
+            set { SetValue(DirectionProperty, value); }
+        }
+
+        public static readonly BindableProperty PopupContentProperty =
+            BindableProperty.Create(nameof(PopupContent), typeof(View), typeof(PopupBehaviour));
+
+        public View PopupContent
```

Comment:
Try to add this as a `ContentProperty` so consumers don't need to do a :

```xml
...
<Popup:PopupBehaviour>
    <Popup:PopupBehaviour.PopupContent>
        ...
     </Popup:PopupBehaviour.PopupContent>
 </Popup:PopupBehaviour>
...
```

But a :
```xml
...
<Popup:PopupBehaviour>
    ...
</Popup:PopupBehaviour>
...
```

Reply:
`Popup Content` can be rename to `Content`?